pkg/functionconfig/scrubber.go

```diff
@@ -225,6 +230,52 @@ func decodeSecretKey(secretKey string) (string, error) {
 	return string(decodedFieldPath), nil
 }
 
-func generateSecretKey(fieldPath string) string {
+func (s *Scrubber) generateSecretKey(fieldPath string) string {
 	return fmt.Sprintf("%s%s", ReferencePrefix, strings.ToLower(fieldPath))
 }
+
+func (s *Scrubber) validateReference(functionConfig *Config,
+	existingSecretMap map[string]string,
+	fieldPath,
+	secretKey,
+	stringValue string) error {
+
+	// for flex volume access keys, we need to check if the volume secret exists
+	if strings.Contains(stringValue, "flexvolume") {
+
+		// get the volume name
+		volumeIndexStr := strings.Split(strings.Split(stringValue, "[")[1], "]")[0]
+		volumeIndex, err := strconv.Atoi(volumeIndexStr)
+		if err != nil {
+			return errors.Wrap(err, "Failed to parse volume index")
+		}
+		volumeName := functionConfig.Spec.Volumes[volumeIndex].Volume.Name
+
+		// list secrets with the volume name label selector
+		volumeSecrets, err := s.KubeClientSet.CoreV1().Secrets(functionConfig.Meta.Namespace).List(context.Background(),
+			metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", common.NuclioResourceLabelKeyVolumeName, volumeName),
+			})
+		if err != nil {
+			return errors.Wrap(err, "Failed to list volume secrets")
+		}
+
+		// if no secret exists, return an error
+		if len(volumeSecrets.Items) == 0 {
+			return errors.New(fmt.Sprintf("No secret exists for volume %s", volumeName))
+		}
+
+		return nil
+	}
+
+	// for other fields, we need to check if the secret exists in the secret map
+	if existingSecretMap != nil {
```

Comment:
else

Reply:
No need actually, 'cause the first if can only end in `return`.